spec/models/request_spec.rb

```diff
@@ -23,32 +23,43 @@
   # so test here on the model. Maybe replace with other controller tests
   # when we work that ticket.
   describe "status enum" do
-    subject(:travel_request) { FactoryBot.build(:travel_request) }
+    let(:staff_profile) { FactoryBot.create(:staff_profile, :with_department) }
+    let(:travel_request) { FactoryBot.build(:travel_request, creator: staff_profile) }
+    let(:absence_request) { FactoryBot.build(:absence_request, creator: staff_profile) }
     it "set expected values" do
-      travel_request.pending!
       expect(travel_request.pending?).to eq true
-      travel_request.approved!
-      expect(travel_request.approved?).to eq true
-      travel_request.denied!
-      expect(travel_request.denied?).to eq true
-      travel_request.changes_requested!
+      travel_request.change_request(agent: staff_profile.department.head)
       expect(travel_request.changes_requested?).to eq true
-      travel_request.canceled!
+      travel_request.fix_requested_changes(agent: absence_request.creator)
+      expect(travel_request.pending?).to eq true
+      travel_request.approve(agent: staff_profile.department.head)
+      expect(travel_request.approved?).to eq true
+      travel_request.cancel(agent: absence_request.creator)
       expect(travel_request.canceled?).to eq true
-      travel_request.recorded!
-      expect(travel_request.recorded?).to eq true
-      travel_request.pending_cancelation!
-      expect(travel_request.pending_cancelation?).to eq true
+    end
+
+    it "can be denied" do
+      travel_request.deny(agent: staff_profile.department.head)
+      expect(travel_request.denied?).to eq true
+    end
+
+    it "transitions through stages for absence" do
+      absence_request.approve(agent: staff_profile.department.head)
```

Comment:
We know that some department heads want to approve absences and others would prefer that the direct supervisor do this. Do we have enough tests here to cover both scenarios? Is it MVP even? We are looking at #35 , but it seems to only apply to travel (which always requires dept. head approval). Do we need to separate this issue out into two, since there are different scenarios for Absences depending on the department?

Reply:
@sdellis at this point I really am not limiting who can approve an absence.  It could be anyone.  I'll add in a check to make sure it is the person's supervisor, or department head.  And I'll add another test.